src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackSerializer.cs

```diff
@@ -462,58 +467,171 @@ private static int LZ4Operation(in ReadOnlySequence<byte> input, Span<byte> outp
         {
             if (msgpackUncompressedData.Length < LZ4NotCompressionSize)
             {
                 writer.WriteRaw(msgpackUncompressedData);
             }
             else
             {
-                var maxCompressedLength = LZ4Codec.MaximumOutputLength((int)msgpackUncompressedData.Length);
-                var lz4Span = ArrayPool<byte>.Shared.Rent(maxCompressedLength);
-                try
+                if (compression == MessagePackCompression.Lz4Block)
                 {
-                    int lz4Length = LZ4Operation(msgpackUncompressedData, lz4Span, LZ4Codec.Encode);
+                    var maxCompressedLength = LZ4Codec.MaximumOutputLength((int)msgpackUncompressedData.Length);
+                    var lz4Span = ArrayPool<byte>.Shared.Rent(maxCompressedLength);
+                    try
+                    {
+                        int lz4Length = LZ4Operation(msgpackUncompressedData, lz4Span, LZ4CodecEncode);
 
-                    const int LengthOfUncompressedDataSizeHeader = 5;
-                    writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4, LengthOfUncompressedDataSizeHeader + (uint)lz4Length));
-                    writer.WriteInt32((int)msgpackUncompressedData.Length);
-                    writer.WriteRaw(lz4Span.AsSpan(0, lz4Length));
+                        const int LengthOfUncompressedDataSizeHeader = 5;
+                        writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4, LengthOfUncompressedDataSizeHeader + (uint)lz4Length));
+                        writer.WriteInt32((int)msgpackUncompressedData.Length);
+                        writer.WriteRaw(lz4Span.AsSpan(0, lz4Length));
+                    }
+                    finally
+                    {
+                        ArrayPool<byte>.Shared.Return(lz4Span);
+                    }
                 }
-                finally
+                else if (compression == MessagePackCompression.Lz4ContiguousBlock)
                 {
-                    ArrayPool<byte>.Shared.Return(lz4Span);
+                    // Write to [Ext(98:int,int...), bin,bin,bin...]
+                    // Reading count from Sequence<T> is better but not yet do.
+                    var sequenceCount = 0;
+                    foreach (var item in msgpackUncompressedData)
+                    {
+                        sequenceCount++;
+                    }
+
+                    writer.WriteArrayHeader(sequenceCount + 1);
+                    writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4Contiguous, sequenceCount * 5));
+                    {
+                        foreach (var item in msgpackUncompressedData)
+                        {
+                            // force write Int32 block(size = 5).
+                            writer.WriteInt32(item.Length);
```

Comment:
If you stored these lengths as individual array elements instead of within your extension, you could use `writer.Write(item.Length);` here and get a more compressed representation of the number.
You'd only have to change the length of your array header from `sequenceCount + 1` to `1+sequenceCount*2`.

Reply:
The reason for using extension is for judgement at the time of deserialization.
If you use an array, it will be impossible to judge.

Reply:
I'm not suggestion you get rid of the extension. You can keep it. I'm just suggesting your extension not store all those integers. Maybe emit the extension header but with a payload length of 0.